core/server/worker/src/main/java/alluxio/worker/netty/DataServerWriteHandler.java

```diff
@@ -62,38 +70,102 @@
   /** The executor service to run the {@link PacketWriter}s. */
   private final ExecutorService mPacketWriterExecutor;
 
+  /**
+   * Special packets used to pass control information from the I/O thread to the packet writer
+   * thread.
+   * EOF: the end of file.
+   * CANCEL: the write request is cancelled by the client.
+   * ABORT: a non-recoverable error is detected, abort this channel.
+   */
+  private static final ByteBuf EOF = Unpooled.buffer(0);
+  private static final ByteBuf CANCEL = Unpooled.buffer(0);
+  private static final ByteBuf ABORT = Unpooled.buffer(0);
```

Comment:
A similar pattern does not work in DataServerReadHandler. But the pattern used in DataServerReadHandler can be applied here. I didn't change this as this is slightly easier to reason about. But I am open to change it to the pattern used in DataServerReadHandler. It requires some work.